contracts/settling_game/L08_Crypts_Resources.cairo

```diff
@@ -0,0 +1,382 @@
+
+    return (days_accrued, seconds_left_over)
+end
+
+# CLAIM CHECK
+@view
+func check_if_claimable{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256
+) -> (can_claim : felt):
+    alloc_locals
+
+    # FETCH AVAILABLE
+    let (days, _) = days_accrued(token_id)
+
+    # ADD 1 TO ALLOW USERS TO CLAIM FULL EPOCH
+    let (less_than) = is_le(days + 1, 1)
+
+    if less_than == TRUE:
+        return (FALSE)
+    end
+
+    return (TRUE)
+end
+
+###########
+# GETTERS #
+###########
+
+# GET RESOURCE LEVEL
+@view
+func get_resource_level{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256, resource : felt
+) -> (level : felt):
+    let (level) = resource_levels.read(token_id, resource)
+    return (level=level)
+end
+
+# GET COSTS
+@view
+func get_resource_upgrade_cost{range_check_ptr, syscall_ptr : felt*, pedersen_ptr : HashBuiltin*}(
+    resource_id : felt
+) -> (cost : Cost):
+    let (cost) = resource_upgrade_cost.read(resource_id)
+    return (cost)
+end
+
+# GET OUTPUT PER ENViRONMENT
+@view
+func get_output_per_environment{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    environment : felt
+) -> (r_output : felt, r_resource_id : felt):
+    alloc_locals
+
+    # Each environment has a designated resourceId
+    let r_resource_id = 22 + environment  # Environment struct is 1->6 and Crypts resources are 23->28
+
+    if environment == 1:
+        return (EnvironmentProduction.DesertOasis, r_resource_id)
+    end
+    if environment == 2:
+        return (EnvironmentProduction.StoneTemple, r_resource_id)
+    end
+    if environment == 3:
+        return (EnvironmentProduction.ForestRuins, r_resource_id)
+    end
+    if environment == 4:
+        return (EnvironmentProduction.MountainDeep, r_resource_id)
+    end
+    if environment == 5:
+        return (EnvironmentProduction.UnderwaterKeep, r_resource_id)
+    end
+    # 6 - Ember's glow is theo pnly one left
+    return (EnvironmentProduction.EmbersGlow, r_resource_id)
+end
+
+############
+# INTERNAL #
+############
+
+# RETURNS RESOURCE OUTPUT
+func calculate_resource_output{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    days : felt, token_id : Uint256, resource_id : felt, output : felt, legendary : felt
+) -> (value : Uint256):
+    alloc_locals
+
+    # GET RESOURCE LEVEL
+    let (level) = get_resource_level(token_id, resource_id)
+
+    # LEGENDARY MAPS EARN MORE RESOURCES
+    let legendary_multiplier = legendary * LEGENDARY_MULTIPLIER
+
+    let (total_work_generated, _) = unsigned_div_rem(
+        days * level * output * legendary_multiplier, 100
+    )
+
+    # IF LEVEL 0 RETURN NO INCREASE
+    # if level == 0:
+    #     return (production_output)
+    # end
+    # return ((level + 1) * production_output)   # HACK: Just to get this compiling
+
+    return (Uint256(total_work_generated, 0))
```

Comment:
Need to include decimals now so this would become 
`return (Uint256(total_work_generated * 10 ** 18, 0))`

(see AMM branch for reference)

Reply:
Resolved in d91be6d